File: 002/euler2.py
def f(n):
    sum_ = 0
    a, b = 0, 1
    while b <= n:
        if b % 2 == 0:
            sum_ += b
        a, b = b, a + b
    return sum_


if __name__ == '__main__':
    print(f(4000000))

File: 002/hs/euler2.hs
fib :: Int -> Integer
fib = (map fib' [0 ..] !!)
  where fib' 0 = 0
        fib' 1 = 1
        fib' n = fib (n-2) + fib (n-1)


main = print $ sum $ takeWhile (<= 4000000) $ filter even $ map fib [0 ..]
